### PR TITLE
Zero To Crypto-Hero: Creating a simple secret note machine

### PR DESCRIPTION
**Pull request by cfanoulis**
Name: Charalampos Fanoulis

Do you run a club? - No but I believe I qualify as an active member

What relevant experiences do you have to help you write this Jam? - Fromerly headed Hackropolis, have previously written a workshop, I have used JS for a while now

Will you be available on Slack to provide support to Hack Clubbers who run into issues while going through your Jam? - Yes!!

### CODE REVIEW

File: jams/zero-to-crypto-hero.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+### Breakdown
+- Preamble: What is encryption? Where do we see it? How is it different that hashing?
+- Set up: Use the HTML repl.it starter OR a custom github starter repo (I will create, contains solely some bare-bones html structure & a js file)
+- Part 1: Create the necessary html UI (2 inputs & 2 buttons "encrypt"/"decrypt")
+- Part 2: Implement the Caesar cipher in JS
+- Part 3: Implement the actual encryption/decryption operation, showing the results on the page
+- Congrats! You have made your first secret note maker!
+
+- How will you ensure that every outcome of the workshop varies (i.e. how will you give jammers a sense of ownership over their project)? - See customization opportunities. The implemented algorithm is intentionally simple, as to encourage jammers to seek out more secure ciphers and implement them themselves
```
Comment (sarthaktexas) [inline jams/zero-to-crypto-hero.md]: okay, i like this! it's pretty simple

File: jams/zero-to-crypto-hero.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- Why should Hack Clubbers care about this project? - This relates to teenagers' personal experiences (lack of privacy from parents, teachers) and is interded to, in a joking way, rebel against it, by passing around messages only those that know can read. Also, technical teenagers are increasingly privacy-minded, and this can be a jump-start to a deep dive in encryption & security
+- General outline of a club meet doing the Jam:
+  - Preamble is being presented by the leader while setup is being done
```
Comment (sarthaktexas) [inline jams/zero-to-crypto-hero.md:27]: only a greek person would write the word preamble

File: jams/zero-to-crypto-hero.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  - Setup is complete; now everyone works towards creating the HTML UI (can be customized as tey wish), only maing sure everyone uses the same element IDs (to simplify debugging)
+  - JS is written according to the workshop, now everyone should have a working caesar machine!
+  - Students get matched in teams, use a shared secret (the shift used for the cipher) and throw notes around, trying to share secrets only amongst them
+  - Really quick recap of the preamble, suggestions for customization are provided
+  - Meet done, now go get the pizzas!
+- What Club Members will walk away with (both in terms of knowledge and in terms of product) - A surface preview fo the world of cryptography and a working encryptor/decryptor for text messages
+- What makes this workshop fun or interesting for Club Members?
+- What platforms will be supported (i.e. MacOS, Windows, Chromebook, Mobile, Browser, etc)? - All (browser development)
+- How will you allow Club Leaders to add their own project to the Jam presentation (giving them a sense of ownership over the meeting)? - At the end, we can include other successful or innnovative algorithms created by clus around the world
```
Comment (sarthaktexas) [inline jams/zero-to-crypto-hero.md]: I don't like this part. I like what line 30 says, that should be the answer to this question imo.

**Review by sarthaktexas — CHANGES_REQUESTED** [review]
requested changes in the last comment :)

File: jams/zero-to-crypto-hero.md (hunk 3 above)
Comment (cfanoulis) [reply jams/zero-to-crypto-hero.md]: I actually don't understand this; is this supposed to allow club leaders to evlove the jam and share it onwards or just each jam leader can personalize it for their club

File: jams/zero-to-crypto-hero.md (hunk 2 above)
Comment (cfanoulis) [reply jams/zero-to-crypto-hero.md:27]: it was late at night and english wouldnt english

**Review by sarthaktexas — COMMENTED** [review]
I think this is really cool as a workshop, but thinking through this I don't think there is opportunity for customization anywhere and therefore not a jam. Please correct me if you disagree, however. Char, you can slack me, and we can hop on a zoom call to brainstorm if you would like.

File: jams/zero-to-crypto-hero.md (hunk 3 above)
Comment (sarthaktexas) [reply jams/zero-to-crypto-hero.md]: If you still have this question, slack me Char.

**Pull request closed without merging**